### ISSUE TITLE
Using OpenSplice: Could not initialize rcl node: error not set

### ISSUE THREAD

**Issue by firesurfer** [ros2/rclcpp#217]
I fixed my wrong cmake file so the executable links to opensplice (See #216 )
Now I get the following error:
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Could not initialize rcl node: error not set
```

This also happens if I try to run the `talker__rmw_opensplice_cpp` example

**Comment by jacquelinekay** [comment]
I believe executables linked to Opensplice require the `install/setup.bash` file to be sourced before running.

**Comment by firesurfer** [comment]
Yeah that's the case. Thanks!

**Comment by dirk-thomas** [comment]
I got exactly the same error message but I don't agree with closing this. The error message is completely useless. The message should state `OSPL_URI not set` but the code which intends to set the error message doesn't work correctly. As far as I can see the following line is broken: https://github.com/ros2/rcl/blob/a01943973a29ca3699a7c026a5a0a0b4acf01e18/rcl/src/rcl/node.c#L102-L103

**Comment by dirk-thomas** [comment]
Closed with ros2/rmw#68
